test/extensions/filters/network/wasm/wasm_filter_test.cc

```diff
@@ -370,6 +376,97 @@ TEST_P(WasmNetworkFilterTest, StopAndResumeUpstreamViaAsyncCall) {
   EXPECT_NE(callbacks, nullptr);
 }
 
+TEST_P(WasmNetworkFilterTest, PanicOnNewConnection) {
+  if (std::get<0>(GetParam()) == "null") {
+    return;
+  }
+  setupConfig("", "panic");
+  setupFilter();
+  EXPECT_CALL(filter(), log_(spdlog::level::critical, _))
+      .Times(testing::AtMost(1)); // Rust logs on panic.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  EXPECT_EQ(Network::FilterStatus::StopIteration, filter().onNewConnection());
+
+  // Should close both up and down streams.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+}
+
+TEST_P(WasmNetworkFilterTest, PanicOnDownstreamData) {
+  if (std::get<0>(GetParam()) == "null") {
+    return;
+  }
+  setupConfig("", "panic");
+  setupFilter();
+  EXPECT_CALL(filter(), log_(spdlog::level::critical, _))
+      .Times(testing::AtMost(1)); // Rust logs on panic.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  Buffer::OwnedImpl fake_downstream_data("Fake");
+  filter().onCreate(); // Create context without calling OnNewConnection.
+  EXPECT_EQ(Network::FilterStatus::StopIteration, filter().onData(fake_downstream_data, false));
+
+  // Should close both up and down streams.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+}
+
+TEST_P(WasmNetworkFilterTest, PanicOnUpstreamData) {
+  if (std::get<0>(GetParam()) == "null") {
+    return;
+  }
+  setupConfig("", "panic");
+  setupFilter();
+  EXPECT_CALL(filter(), log_(spdlog::level::critical, _))
+      .Times(testing::AtMost(1)); // Rust logs on panic.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Open);
+  Buffer::OwnedImpl fake_downstream_data("Fake");
+  filter().onCreate(); // Create context without calling OnNewConnection.
+  EXPECT_EQ(Network::FilterStatus::StopIteration, filter().onWrite(fake_downstream_data, false));
+
+  // Should close both up and down streams.
+  EXPECT_EQ(read_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+  EXPECT_EQ(write_filter_callbacks_.connection().state(), Network::Connection::State::Closed);
+}
+
+TEST_P(WasmNetworkFilterTest, CloseDownstream) {
+  if (std::get<1>(GetParam()) == "rust") {
+    // TODO(mathetake): not yet supported in the Rust SDK.
```

Comment:
This is blocked on https://github.com/proxy-wasm/proxy-wasm-rust-sdk/pull/64, I'll add this once it's merged, so feel free to re-assign to me (if there are other changes, but don't waste CI cycle on it).